ykrt/src/thread_intercept.rs

```diff
@@ -1,11 +1,83 @@
 use std::os::raw::{c_int, c_void};
 
+use std::{ffi::CString, ptr::null_mut};
+
+use libc::{dlsym, free, malloc, pthread_attr_t, pthread_create, pthread_t};
+
+const SHADOW_STACK_SIZE: usize = 1000000;
+
+#[derive(Debug)]
+struct ThreadRoutine {
+    pub args: *mut c_void,
+    pub start_routine: extern "C" fn(*mut c_void) -> *mut c_void,
+}
+
+extern "C" fn wrap_thread_routine(arg: *mut c_void) -> *mut c_void {
+    let str = CString::new("shadowstack_0").unwrap();
+    let shadowstack_symbol_addr = unsafe {
+        //  Obtain address of a shadowstack_0 symbol
+        dlsym(null_mut(), str.as_ptr() as *const i8)
+    };
+    if shadowstack_symbol_addr.is_null() {
+        panic!("Unable to find shadowstack address")
+    }
+    let stack_addr = unsafe {
+        // Allocate stack
```

Comment:
I think we can kill this comment.

Reply:
OK, killed 👉  7bb37f0ab0129a7eed413b2d3a0a0c5b3c950355